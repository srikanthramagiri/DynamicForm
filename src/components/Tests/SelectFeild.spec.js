import {shallow} from 'enzyme'
import SelectFeild from '../FormComponents/SelectFeild'

describe('select feild component', ()=> {
    const optionsValues=  ["India", "srilanka", "Algeria", "Andorra"]
    const changeHandlerSpy = jest.fn();
    const wrapper = shallow(<SelectFeild values={optionsValues} changeHandler={changeHandlerSpy}/>)
    const Select =  wrapper.find(`[data-test='select-feild']`)
    it('should render correctly', ()=> {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper).not.toBeNull()
    })
    it('should have one input text area',()=> {
        expect(Select).not.toBeNull()
    }) 
    it('change handler', ()=> {
        Select.simulate('change', {target: {value: 'india'}})
        expect(changeHandlerSpy).toHaveBeenCalled()
    })
    it('change handler when emty value', ()=> {
        Select.simulate('change', {target: {value: ''}})
        expect(changeHandlerSpy).toHaveBeenCalled()
    })
})