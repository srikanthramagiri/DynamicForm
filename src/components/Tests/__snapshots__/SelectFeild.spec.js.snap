// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`input text field component should render correctly 1`] = `ShallowWrapper {}`;

exports[`input text field should render correctly 1`] = `ShallowWrapper {}`;

exports[`select field component should render correctly 1`] = `ShallowWrapper {}`;
